debezium-core/src/test/java/io/debezium/util/parser/TokenStream.java

```diff
@@ -0,0 +1,2592 @@
+/*
+ * Copyright Debezium Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.debezium.util.parser;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.ListIterator;
+import java.util.NoSuchElementException;
+import java.util.Objects;
+import java.util.function.IntConsumer;
+import java.util.function.LongConsumer;
+
+import io.debezium.annotation.Immutable;
+import io.debezium.annotation.NotThreadSafe;
+import io.debezium.function.BooleanConsumer;
+import io.debezium.text.ParsingException;
+import io.debezium.text.Position;
+import io.debezium.text.XmlCharacters;
+import io.debezium.util.Strings;
+
+/**
+ * A copy of {@link io.debezium.text.TokenStream} class that provides additional methods necessary for test parser.
```

Comment:
Ugh :) Why again do we need `TokenStream` in main? And if it's truly needed over there (and not test-only), can't we extend?

Reply:
It is used in `TableIdParser` and in `Strings`.

I was attemptin to do the extension first but that would need not insignificant changes including even generfication of the class which I thought is unnecessary complication. Hence I followed the path of the least resistence and made the copy so the original tokenizer is not polluted with SQL specific stuff.